src/app/modules/overlay-slide-ins/components/overlay-container/overlay-container.component.ts

```diff
@@ -0,0 +1,35 @@
+import { CdkPortalOutlet } from '@angular/cdk/portal';
+import {
+  ChangeDetectionStrategy, Component, ElementRef, ViewChild,
+} from '@angular/core';
+
+@Component({
+  selector: 'ix-overlay-container',
+  templateUrl: './overlay-container.component.html',
+  styleUrl: './overlay-container.component.scss',
+  standalone: true,
+  imports: [
+    CdkPortalOutlet,
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class OverlayContainerComponent {
+  @ViewChild(CdkPortalOutlet, { static: true }) portalOutlet!: CdkPortalOutlet;
+  constructor(private host: ElementRef<HTMLElement>) {}
+
+  startCloseAnimation(): Promise<void> {
```

Comment:
Can we not use Angualr's :enter and :leave animation instead of doing things manually?